mtx/public/mtx_solve_routines.inc

```diff
@@ -282,7 +283,8 @@
          real(dp) :: temp
          integer :: i,k
          info = 0
-         do i = 1, n
+         !$omp simd private(temp)
```

Comment:
Follow the general strategy of `!$OMP PARALLEL` outer loop and `!$omp simd` inner loop